### PR TITLE
Dynamic scripts

### PR DESCRIPTION
**Pull request by manics**
developers/scripts: Add section on `NSDYNAMIC`

See:
- openmicroscopy/openmicroscopy#3790
- https://trello.com/c/JUlWkFvS/48-are-dynamic-scripts-documented#comment-56c2efbaea9e1dbe601b1b34

If you want to be particularly throrough download and upload the script to OMERO, and check that everytime you run it it shows the current list of datasets (which may have been added/removed in the intervening time)

### CODE REVIEW

File: omero/developers/scripts/user-guide.txt [hunk 1]
```diff
@@ -98,6 +98,20 @@ and the script returns them as outputs.
     # Cleanup
     client.closeSession()
 
+Dynamic arguments
+^^^^^^^^^^^^^^^^^
+
+In general, script parameters are processed on server startup
+and cached for later use. If you have a script which should
+re-calculate its arguments on each invocation, use the
```
Comment (mtbc) [inline omero/developers/scripts/user-guide.txt]: "recalculate"

**Comment by manics** [comment]
@mtbc Typo fixed

**Comment by emilroz** [comment]
Looks good to me. The example script works as expected.

screen shot 2016-02-24 at 11 27 23

**Comment by pwalczysko** [comment]
Uploaded the script to eel merge.
The changes in datasets were respected every time the script was launched.

After first edit of a name of Dataset 6901
screen shot 2016-02-24 at 10 49 20
After second edit of this name 
screen shot 2016-02-24 at 10 49 55
After deletion of the dataset:
screen shot 2016-02-24 at 10 50 28

The only minor issue is : Why did the edit change the order in the list ? (originally ordered by ID, after edit of name, the edited dataset was listed the last, no reason for that)

I suppose that the minor issue is not blocking to this PR as this is just an example. All seems good here.

**Comment by emilroz** [comment]
@pwalczysko I think (@manics can correct me if I'm wrong) the order of the Datasets in this list is not created by the script but by the server and it's not guaranteed to be always the same so in a sense it's random.

**Comment by manics** [comment]
It's whatever is returned by `conn.getObjects('Dataset')`, and since this is an example of dynamic parameters aimed at developers I think it's better to minimise the extraneous code.

**Comment by pwalczysko** [comment]
@manics @emilroz : Thank you for the clarification.

**Comment by hflynn** [comment]
Is this good to merge then?

**Comment by pwalczysko** [comment]
Fine by me.

**Pull request merged**